gradio/processing_utils.py

```diff
@@ -275,6 +277,8 @@ def _move_to_cache(d: dict):
                         f"File {path} is not in the upload folder and cannot be accessed."
                     )
             temp_file_path = block.move_resource_to_block_cache(payload.path)
+            if keep_in_cache:
```

Comment:
We should only be adding files to this set, right now we add aanydata (pd.DataFrame, strings etc) if `keep_in_cache` is True

Reply:
Hmm how do you figure that? `_move_to_cache` is only called if the value is a `FileData` dictionary, meaning that `temp_file_path` must be a string path.

Reply:
I moved it to the end of `_move_to_cache` to be more explicit about this

Reply:
Yea you're right I was mistaken! Should the predicate in `traverse` used here be `file_obj_with_meta` though?

Reply:
Yes, I think so. Let me just make sure that doesn't break older apps

Reply:
Actually unfortunately it does seem to break `gr.load()` with older Spaces that have examples because the format of file data in those examples doesn't have "meta". We may need to able support both. I'll see if there's a clean way to fix this as part of tackling https://github.com/gradio-app/gradio/issues/7362.